### DISCUSSION TITLE
Issues with initiating new pod

### DISCUSSION THREAD

**Discussion by RagingPuppies** [linkerd/linkerd2#4485] [General]
Hi,
i've seen on many of my clusters this issue but have'nt seen anything related in google\github
i use HA linkerd2 and my pods from time to time stay in notready state, the log in the proxy sidecar:

> time="2020-05-27T10:41:23Z" level=info msg="running version stable-2.6.0"5/27/2020 11:41:23 AM INFO [ 0.003103s] linkerd2_proxy::app::main using destination service at ControlAddr { addr: Name(NameAddr { name: "linkerd-dst.linkerd.svc.cluster.local", port: 8086 }), identity: Some("linkerd-destination.linkerd.serviceaccount.identity.linkerd.cluster.local") }5/27/2020 11:41:23 AM INFO [ 0.003140s] linkerd2_proxy::app::main using identity service at Name(NameAddr { name: "linkerd-identity.linkerd.svc.cluster.local", port: 8080 })5/27/2020 11:41:23 AM INFO [ 0.003145s] linkerd2_proxy::app::main routing on V4(127.0.0.1:4140)5/27/2020 11:41:23 AM INFO [ 0.003150s] linkerd2_proxy::app::main proxying on V4(0.0.0.0:4143) to None5/27/2020 11:41:23 AM INFO [ 0.003156s] linkerd2_proxy::app::main serving admin endpoint metrics on V4(0.0.0.0:4191)5/27/2020 11:41:23 AM INFO [ 0.003158s] linkerd2_proxy::app::main protocol detection disabled for inbound ports {25, 587, 3306}5/27/2020 11:41:23 AM INFO [ 0.003174s] linkerd2_proxy::app::main protocol detection disabled for outbound ports {25, 587, 3306}5/27/2020 11:41:23 AM ERR! [ 0.021911s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:41:33 AM ERR! [ 10.027362s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:41:43 AM ERR! [ 20.033154s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:41:53 AM ERR! [ 30.038131s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:42:03 AM ERR! [ 40.043742s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:42:13 AM ERR! [ 50.049944s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:42:23 AM ERR! [ 60.056555s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:42:33 AM ERR! [ 70.061260s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet5/27/2020 11:42:43 AM ERR! [ 80.067105s] admin={bg=identity} linkerd2_proxy::app::identity Received invalid ceritficate: invalid certificate: CertNotValidYet

and it can stay like that for long time,
sometimes when i redeploy the pod it rise normal and sometimes not.
how can i debug this issue?
Thanks

**Comment by grampelberg** [comment]
`CertNotValidYet` suggests that you have an issue with timing between your nodes. The server running `linkerd-identity` is likely ahead of the node that is running the pod with that certificate.

**Reply by sriveros95** [reply]
@grampelberg What could one do in this case?

**Reply by grampelberg** [reply]
Setup [ntp](http://www.ntp.org/) on the nodes.
